src/server/main_service.cc

```diff
@@ -634,6 +634,7 @@ void Service::DispatchCommand(CmdArgList args, facade::ConnectionContext* cntx)
 
   uint64_t start_usec = ProactorBase::GetMonotonicTimeNs(), end_usec;
 
+  DispatchMonitorIfNeeded(cid->opt_mask() & CO::ADMIN, dfly_cntx, args);
```

Comment:
why moving the command solves the issue?

Reply:
because otherwise it would hae a chance to execute it, The pytest show that its sending the command over the monitor.